packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -200,25 +228,33 @@ abstract class ShortcutActivator {
   /// this is only used to query whether [HardwareKeyboard.logicalKeysPressed]
   /// contains a key.
   ///
-  /// Since [ShortcutActivator] accepts all event types, subclasses might want
-  /// to check the event type in [accepts].
-  ///
   /// See also:
   ///
   ///  * [LogicalKeyboardKey.collapseSynonyms], which helps deciding whether a
   ///    modifier key is pressed when the side variation is not important.
-  bool accepts(RawKeyEvent event, RawKeyboard state);
+  bool acceptsEvent(KeyEvent event, HardwareKeyboard state);
 
-  /// Returns true if the event and keyboard state would cause this
+  /// Returns true if the event and current [RawKeyboard] state would cause this
   /// [ShortcutActivator] to be activated.
   ///
-  /// If the keyboard `state` isn't supplied, then it defaults to using
-  /// [RawKeyboard.instance].
+  /// **This method is DEPRECATED.** prospective callers should call
+  /// [isActivatedByEvent] instead.
+  @Deprecated(
+    'Call isActivatedByEvent instead. '
+    'This feature was deprecated after v3.16.0-13.0.pre.',
+  )
   static bool isActivatedBy(ShortcutActivator activator, RawKeyEvent event) {
     return (activator.triggers?.contains(event.logicalKey) ?? true)
         && activator.accepts(event, RawKeyboard.instance);
   }
 
+  /// Returns true if the event and [HardwareKeyboard] state would cause this
+  /// [ShortcutActivator] to be activated.
+  static bool isActivatedByEvent(ShortcutActivator activator, KeyEvent event) {
+    return (activator.triggers?.contains(event.logicalKey) ?? true)
+        && activator.acceptsEvent(event, HardwareKeyboard.instance);
```

Comment:
I've been thinking about `triggers` since adding it and honestly I kind of regret it. It was added as an optimization before benchmarking was done. It forbids users to index activators in their own way, and makes logic such as `isActivatedByEvent` complicated instead of being able to simply use `acceptsEvent`.

I think we can take this opportunity to redefine `acceptsEvent` to include `triggers`, so that we don't need `isActivatedByEvent` at all; and we redefine `triggers` as an optional hint to accelerate looking up. What do you think?

Reply:
I like the idea, and I see why you would want that.  I'm worried that this will cause a lot of churn in the API and make it harder to migrate.  If we made this change, we'd probably also want to deprecate `triggers` and `isActivatedByEvent` in the API, which makes for a pretty big change.

But I guess we're already considering just making a hard break for the switch to `KeyEvent` here, so maybe now is the time to do it all.

Reply:
Oh, sorry, I misunderstood, I think.  I thought you were suggesting adding triggers as an optional argument to `accepts`, but I think you were just saying to change the meaning of it, but leave it as a getter on the class.  That's a smaller change, which I like better.